### PR TITLE
FAI-12391 | Added permissions required for each GitHub stream

### PR DESCRIPTION
**Pull request by dbruno21**
## Description

Added permissions required for each GitHub stream

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/github-source/src/org-repo-filter.ts [hunk 1]
```diff
@@ -100,10 +97,7 @@ export class OrgRepoFilter {
         this.filterConfig.reposByOrg.get(org).forEach((repoName) => {
           const repo = visibleRepos.find((r) => r.name === repoName);
           if (!repo) {
-            this.logger.warn(
-              `Skipping not found repository ${org}/${repoName}`
-            );
-            return;
+            throw new VError(`Repository not found: ${org}/${repoName}`);
```
Comment (dbruno21) [inline sources/github-source/src/org-repo-filter.ts:100]: I see on feeds there's a setting for this `fail_missing_repos`, should we add (on another PR)? @chalenge

Comment (dbruno21) [reply sources/github-source/src/org-repo-filter.ts]: reverted

File: sources/github-source/src/org-repo-filter.ts [hunk 2]
```diff
@@ -63,19 +64,15 @@ export class OrgRepoFilter {
...
       if (!this.filterConfig.organizations) {
+        const visibleOrgs = await github.getOrganizations();
         visibleOrgs.forEach((org) => {
           if (!this.filterConfig.excludedOrganizations?.has(org)) {
             organizations.add(org);
           }
         });
       } else {
         this.filterConfig.organizations.forEach((org) => {
-          if (!visibleOrgs.some((o) => o === org)) {
```
Comment (dbruno21) [inline sources/github-source/src/org-repo-filter.ts:75]: Fine-grained tokens return an empty list when listing organizations for the authenticated user, so we can't validate against visible orgs and organizations in that case.

Comment (dbruno21) [reply sources/github-source/src/org-repo-filter.ts:75]: Left check agains visible orgs only if visible orgs list is not empty.

File: sources/github-source/src/org-repo-filter.ts [hunk 3]
```diff
@@ -72,7 +73,11 @@ export class OrgRepoFilter {
...
+          // fine-grained tokens return an empty list for visible orgs,
+          // so we only run the check if the list is not empty
+          if (visibleOrgs.length === 0) {
```
Comment (chalenge) [inline sources/github-source/src/org-repo-filter.ts]: `visibleOrgs.length` works too.

File: sources/github-source/src/org-repo-filter.ts [hunk 4]
```diff
@@ -64,6 +65,9 @@ export class OrgRepoFilter {
       const organizations = new Set<string>();
       const github = await GitHub.instance(this.config, this.logger);
       const visibleOrgs = await github.getOrganizations();
+      if (visibleOrgs.length === 0) {
```
Comment (chalenge) [inline sources/github-source/src/org-repo-filter.ts:68]: `if visibleOrgs.length` works fine

File: sources/github-source/src/org-repo-filter.ts (hunk 2 above)
Comment (chalenge) [reply sources/github-source/src/org-repo-filter.ts:75]: Another qst, with fine-grained token does GET `/organization/{orgId}` work then?

File: sources/github-source/README.md [hunk 5]
```diff
@@ -79,10 +79,18 @@ connectors, such as linting/formatting tools, should go in the root
...
 
-| Stream        | Classic               | Fine-grained*                            |
-|---------------|-----------------------|------------------------------------------|
-| Commits       | repo:status, read:org | Repository Contents, Repository Metadata |
-| Copilot Seats | read:org              | Organization Administration              |
-| Copilot Usage | read:org              | Organization Administration              |
-
-*Fine-grained permissions marked as required are always read-only.
\ No newline at end of file
+*Fine-grained permissions marked as required are always read-only.
```
Comment (chalenge) [inline sources/github-source/README.md:82]: Can we add the comment about fine-grained permission only applying to one organization

**Review by chalenge — APPROVED** [review]
lgtm, after comment

**Pull request merged**